lib/workers/repository/init/merge.ts

```diff
@@ -341,24 +342,100 @@ export function setNpmTokenInNpmrc(config: RenovateConfig): void {
   delete config.npmToken;
 }
 
-export async function mergeStaticRepoEnvConfig(
+export async function resolveStaticRepoConfig(
   config: AllConfig,
   env: NodeJS.ProcessEnv,
 ): Promise<AllConfig> {
+  let fileConfig: AllConfig | undefined;
+
+  try {
+    fileConfig = await tryReadStaticRepoFileConfig(
+      env.RENOVATE_STATIC_REPO_CONFIG_FILE,
+    );
+  } catch (err) {
+    logger.fatal({ err }, 'Failed to load static repository config file');
+    process.exit(1);
+  }
+
+  const envConfig = await tryLoadStaticRepoEnvConfig(env);
+
+  const staticRepoConfig = fileConfig ?? envConfig;
+
+  if (fileConfig && envConfig) {
+    logger.warn(
+      { fileConfig, envConfig },
+      'Both file and env static repository configs found — using file config',
+    );
+  }
+
+  if (is.nullOrUndefined(staticRepoConfig)) {
+    return config;
+  }
+
+  return mergeStaticConfig(config, staticRepoConfig);
+}
+
+export async function tryReadStaticRepoFileConfig(
+  staticRepoConfigFile: string | undefined,
+): Promise<AllConfig | undefined> {
+  if (!is.nonEmptyString(staticRepoConfigFile)) {
+    return undefined;
+  }
+
+  logger.debug({ staticRepoConfigFile }, 'reading static repo config file');
+
+  let staticRepoConfigRaw: string | undefined;
+  try {
+    staticRepoConfigRaw = await readSystemFile(staticRepoConfigFile, 'utf8');
+  } catch (err) {
+    throw new Error(
+      `Failed to read static repo config file: "${staticRepoConfigFile}"`,
+      { cause: err },
+    );
+  }
+
+  const parsed = parseJson(
+    staticRepoConfigRaw,
+    staticRepoConfigFile,
+  ) as AllConfig;
+  const staticRepoConfig = await migrateAndValidateConfig(
+    parsed,
+    staticRepoConfigFile,
+  );
```

Comment:
What's the result if the config is invalid? e.g. throws an exception and exits, or ignores it and continues?

Reply:
> !!! warning If the file is missing or contains invalid configuration, the scan will be aborted.

we mentioned it in the md and its unit tested as well, we abort scan with process exit.